src/main/java/gregtech/common/covers/CoverFluidRegulator.java

```diff
@@ -315,12 +334,20 @@ public void writeToNBT(NBTTagCompound tagCompound) {
         super.writeToNBT(tagCompound);
         tagCompound.setInteger("TransferMode", transferMode.ordinal());
         tagCompound.setInteger("TransferAmount", transferAmount);
+        tagCompound.setTag("filterv2", new NBTTagCompound());
     }
 
     @Override
     public void readFromNBT(NBTTagCompound tagCompound) {
         super.readFromNBT(tagCompound);
         this.transferMode = TransferMode.values()[tagCompound.getInteger("TransferMode")];
+        //legacy NBT tag
+        if (!tagCompound.hasKey("filterv2") && tagCompound.hasKey("TransferAmount")) {
```

Comment:
I Notice that this is the only cover with this legacy check, but do the new covers even serialize with this Tag? I don't see them overriding the Pump/Conveyor write/read toNBT methods, nor do I see any changes in the Fluid Filter serialization tag names

Reply:
yeah this is actually only needed on the fluid regulator, since we dont have to migrate the voiding covers yet...
since this PR changes the behavior of already placed fluid filters, this tag is only used to mark if the amount previously set on the regulator as an amount to supply/ extract exact will also be set to the configured fluids on the filter